packages/@aws-cdk/aws-stepfunctions/lib/state-graph.ts

```diff
@@ -159,4 +161,35 @@ export class StateGraph {
     }
   }
 
+  /**
+   * Binds this StateGraph to the StateMachine it defines and updates state machine permissions
+   */
+  public bind(stateMachine: StateMachine) {
+
+    for (const state of this.allStates) {
+      if (DistributedMap.isDistributedMap(state)) {
+        stateMachine.addToRolePolicy(new iam.PolicyStatement({
+          actions: ['states:StartExecution'],
+          resources: [stateMachine.stack.formatArn({
+            service: 'states',
+            resource: 'stateMachine',
+            resourceName: '*',
```

Comment:
Not loving the wildcard here.

According to the docs we should be able to lock this down.
https://docs.aws.amazon.com/step-functions/latest/dg/concepts-asl-use-map-state-distributed.html#distributed-map-iam-policies

However I can see us running into a circular dependency here. Did you explore this?

Reply:
Yes.  I originally tried using `addRoleToPolicy` with the state machine name and then I tried using the `grantStartExecution` method, but I did indeed run into a circular dependency.  I wasn't completely sure how to proceed.

I noticed that the state machine adds an explicit dependency on the role and doing so also adds an explicit dependency on the role's policy to the state machine.  We shouldn't need an explicit dependency on the policy since in the resulting CloudFormation template both the state machine and the policy depend on the role.

I tried removing the explicit `stateMachine.addDependency` call where the role gets added as a dependency to the state machine.  This approach worked, but had huge changes across integration tests throughout the CDK library.  I decided it was best not to update all those integration tests for this one use case.

So, in short, IMO the CDK is adding explicit dependencies where they are not needed and fixing that would fix the circular dependency issue and allow us to lock this down.  But, changing that behavior would be a significant change across the repository, not just affecting the stepfunctions library.